ami/trainers/forward_dynamics_trainer.py

```diff
@@ -0,0 +1,95 @@
+from functools import partial
+
+import torch
+from torch.distributions import kl_divergence
+from torch.distributions.normal import Normal
+from torch.nn.functional import mse_loss
+from torch.optim import Optimizer
+from torch.utils.data import DataLoader
+
+from ami.data.buffers.buffer_names import BufferNames
+from ami.data.buffers.causal_data_buffer import CausalDataBuffer
+from ami.data.buffers.random_data_buffer import RandomDataBuffer
+from ami.data.interfaces import ThreadSafeDataUser
+from ami.models.components.forward_dynamics import ForwardDynamics
+from ami.models.model_names import ModelNames
+from ami.models.model_wrapper import ModelWrapper
+from ami.models.vae import VAE, Decoder, Encoder
+
+from .base_trainer import BaseTrainer
+
+
+class ForwardDynamicsTrainer(BaseTrainer):
+    def __init__(
+        self,
+        partial_dataloader: partial[DataLoader[torch.Tensor]],
+        partial_optimizer: partial[Optimizer],
+        device: torch.device,
+        observation_encoder_name: ModelNames | None = None,
+        max_epochs: int = 1,
+        minimum_dataset_size: int = 1,
```

Comment:
データセット取得の実装の都合上、minimum_dataset_sizeは2以上でなくてはなりません。
できれば`assert minimum_dataset_size >= 2`を入れておきたいですね。
```suggestion
        minimum_dataset_size: int = 2,
```

Reply:
対応しました https://github.com/MLShukai/ami/pull/123/commits/2120115ddbb02186222fb03b8f4713012ed24e9b